source/grid/tria.cc

```diff
@@ -14674,14 +14699,35 @@ Triangulation<dim, spacedim>::prepare_coarsening_and_refinement()
 
                                     // step 1: detect, along which axis the face
                                     // is currently refined
+
+                                    // first, we need an iterator pointing to
+                                    // the parent face. This requires a slight
+                                    // detour in case the neighbor is behind a
+                                    // periodic face.
+                                    const auto parent_face = [&]() {
+                                      if (has_periodic_neighbor)
+                                        {
+                                          const auto &neighbor =
+                                            cell->periodic_neighbor(i);
+                                          const auto parent_face_no =
+                                            neighbor
+                                              ->periodic_neighbor_of_periodic_neighbor(
+                                                nb_indices.first);
+                                          auto parent =
+                                            neighbor->periodic_neighbor(
+                                              nb_indices.first);
+                                          return parent->face(parent_face_no);
+                                        }
+                                      else
+                                        return cell
+                                          ->neighbor_or_periodic_neighbor(i)
```

Comment:
you're in the `else` branch for `if (has_periodic_neighbor)`, so you can probably just call `cell->neighbor(i)` here